### PR TITLE
Enable POM structure and libyui for firstboot

### PR DESCRIPTION
**Pull request by JRivrain**
4f74e655aaedad31d2e665c267aa15b6024415cf: Enable POM structure and libyui for firstboot

Create initial POM structure
Enable libyui in both autoyast and regular firstboot tests
Introduce asserting pages with debug_label rather than with random page elements (not working yet, see https://progress.opensuse.org/issues/89638 and https://progress.opensuse.org/issues/89866)

Needles: https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/716
- Verification run: 
[TW-firstboot](http://waaa-amazing.suse.cz/tests/14988) [TW-fisrtboot-custom](http://waaa-amazing.suse.cz/tests/14973) [TW-firstboot-autoyast](http://waaa-amazing.suse.cz/tests/14985) [LEAP-firstboot-autoyast](http://waaa-amazing.suse.cz/tests/14986) [LEAP-firstboot](http://waaa-amazing.suse.cz/tests/14976)
[SLE-firstboot](http://waaa-amazing.suse.cz/tests/14977) [SLE-firstboot-custom](http://waaa-amazing.suse.cz/tests/14978) [SLE-firstboot-autoyast](http://waaa-amazing.suse.cz/tests/14987) [SLE-firstboot-textmode](http://waaa-amazing.suse.cz/tests/14980)

### CODE REVIEW

**Review by jknphy — COMMENTED** [review]
LGTM, some minor things, comments or some variables to remove.

Feel free (optionally) if you want to offer more granularity, at least I see 3 commits, (1) ay profiles/schedules (2) controller/pages and other rest api stuff (3) tests. But it is not really necessary.

File: tests/installation/enable_y2_firstboot.pm [hunk 1]
```diff
@@ -18,6 +18,7 @@ use testapi;
 use utils qw(zypper_call clear_console);
 use scheduler 'get_test_suite_data';
 
+
```
Comment (rwx788) [inline tests/installation/enable_y2_firstboot.pm:21]: I guess we also can revert this change.

File: tests/installation/setup_libyui_firstboot.pm [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+use warnings;
+use base "installbasetest";
+use testapi;
+
+use testapi;
+use registration "add_suseconnect_product";
+use version_utils "is_sle";
+
+sub run {
+    add_suseconnect_product('sle-module-development-tools') if is_sle;
```
Comment (rwx788) [inline tests/installation/setup_libyui_firstboot.pm:31]: I wonder if we should use same approach @OleksandrOrlov has used for the yast2 gui tests, and we also should clean up registration before enabling new modules in order not to affect other jobs which use same image. So scheduling `scc_cleanup_reregister` would help there.

**Review by rwx788 — COMMENTED** [review]
Looks very promising, couple of very minor points.

File: tests/installation/setup_libyui_firstboot.pm (hunk 2 above)
Comment (rwx788) [reply tests/installation/setup_libyui_firstboot.pm:31]: Actually, let's leave it for now as we have https://progress.opensuse.org/issues/90368

File: lib/YuiRestClient.pm [hunk 3]
```diff
@@ -135,4 +135,14 @@ sub set_libyui_backend_vars {
     set_var('YUI_SERVER', $server);
 }
 
+sub setup_libyui_firstboot {
+    my $port = get_var('YUI_PORT');
```
[the remark this answers is not in the document]
Comment (JRivrain) [reply lib/YuiRestClient.pm:139]: > I would suggest to remove module 'check_libyui_firstboot.pm' and move `YuiRestClient::connect_to_app()` to 'yast2_firstboot' module.

I prefer this way, because if you see the test failing, you can pause it, check the "details" tab to see what port is used and connect quickly with VNC. Otherwise, you don't know what port is used (though, you could deduct it) as the module did not pass yet. It cannot be checked earlier as libyui is not yet active before reboot.

[the remark this answers is not in the document]
Comment (jknphy) [reply lib/YuiRestClient.pm:139]: you can still display both [here](http://waaa-amazing.suse.cz/tests/14988#step/setup_libyui_firstboot/7) (which is before than where is set now) and the connection just before acting on the UI. Am I missing something?

[the remark this answers is not in the document]
Comment (JRivrain) [reply lib/YuiRestClient.pm:139]: Well we would need to start a module to start libyui at that point no ? We could just show the port, but we would miss the info if libyui is broken after reboot, so it would be harder to troubleshoot in such case.

[the remark this answers is not in the document]
Comment (JRivrain) [reply lib/YuiRestClient.pm:139]: Ah, sorry, now I understand what you mean

[the remark this answers is not in the document]
Comment (JRivrain) [reply lib/YuiRestClient.pm:139]: So I applied what you suggested.

**Comment by JRivrain** [comment]
> LGTM, some minor things, comments or some variables to remove.
> 
> Feel free (optionally) if you want to offer more granularity, at least I see 3 commits, (1) ay profiles/schedules (2) controller/pages and other rest api stuff (3) tests. But it is not really necessary.

I agree it would be nicer, but a bit complicated at that point.

**Review by rwx788 — COMMENTED** [review]
I believe we are good to merge and keep introduce further improvements.

**Review by jknphy — APPROVED** [review]

**Comment by DimStar77** [comment]
https://openqa.opensuse.org/tests/1696355#next_previous

This fails in TW (makes me wonder about the verification runs executed, but TW also moved from libyui 4.1 to 4.2)

so, the question basically is: test bug or product bug?

**Comment by rwx788** [comment]
> https://openqa.opensuse.org/tests/1696355#next_previous
> 
> This fails in TW (makes me wonder about the verification runs executed, but TW also moved from libyui 4.1 to 4.2)
> 
> so, the question basically is: test bug or product bug?

There is variable missing, we will take care of fixing this. Sorry for the noise, I'm testing the patch here: https://openqa.opensuse.org/tests/1696774

**Comment by DimStar77** [comment]
> There is variable missing, we will take care of fixing this. Sorry for the noise, I'm testing the patch here: https://openqa.opensuse.org/tests/1696774

Thank you! I tagged three test fails with poo#89866 - if your fix/test run works, I'd appreciate you adjusting the variables as needed for all the ones tagged

**Comment by JRivrain** [comment]
Still failing for autoyast, however if I clone it to my workstation, it works. I could not find any difference between the two, same code, same variables, same profile :'( . Anyway, not a product bug as it passed for me also in the last build: http://waaa-amazing.suse.cz/tests/15010.
But on o3, it looks like the init script does not work:
o3_nofirstboot
